projects/laji/src/app/shared-modules/select/select/select.component.ts

```diff
@@ -189,17 +238,35 @@ export class SelectComponent implements OnInit, OnChanges, OnDestroy {
     }
     this.selectedOptions = [];
     if (!selected || selected.length === 0) {
+      this.options.map(option => {
+        option.checkboxValue = this.checkboxType === 'basic' ? false : undefined;
+      });
       this.unselectedOptions = this.options;
       return;
     }
     this.unselectedOptions = [];
+    const countObj = this.selected !== undefined ?
+    this.selected.some((item: any) => item !== undefined && (item.checkboxValue === true || item.checkboxValue === false)) : false;
+
     this.options.map(option => {
-      if (selected.includes(option.id)) {
-        this.selectedOptions.push(option);
+      if (countObj) {
+        selected.map(item => {
+          if (item.id === option.id) {
+            this.selectedOptions.push({'id': option.id, 'value': option.value, 'info': option.info, 'checkboxValue': item.checkboxValue});
+          } else {
+            this.unselectedOptions.push({'id': option.id, 'value': option.value, 'info': option.info, 'checkboxValue': item.checkboxValue});
+          }
+        });
       } else {
-        this.unselectedOptions.push(option);
+        if (selected.includes(option.id)) {
+          this.selectedOptions.push({'id': option.id, 'value': option.value, 'info': option.info, 'checkboxValue': true});
+        } else {
+          this.unselectedOptions.push({'id': option.id, 'value': option.value, 'info': option.info, 'checkboxValue': false});
+        }
       }
+
```

Comment:
countObj could be replaced with something more descriptive. hmmm I think lines 248 - 268 could be replaced with somethings like this:
```
    this.options.forEach(option => {
      const selectedItem = selected.find(select =>
        (option.id === select) ||
        (option.id === select?.id && (select.checkboxValue === true || select.checkboxValue === false))
      );
      const targetOptions = selectedItem ? this.selectedOptions : this.unselectedOptions;
      const checkboxValue = typeof selectedItem === 'object' && 'checkboxValue' in selectedItem ? selectedItem.checkboxValue : !!selectedItem;

      targetOptions.push({
        ...option,
        checkboxValue
      });
    });